kernelci/config/build.py

```diff
@@ -239,6 +221,16 @@ def cc(self):
     def cc_version(self):
         return self._cc_version
 
+    @property
+    def arch_params(self):
+        return self._arch_params.copy()
```

Comment:
I don't quite understand a copy of the attribute is returned here? returning a copy of an object with a `.` operator seems a bit counter intuitive.

Reply:
That's to ensure the client code can't change the parameters.  If we return the dictionary object reference then it's not read-only any more.

Reply:
The property is meant to be read-only afaik.  There could be a setter property instead if we wanted this to be mutable.

Reply:
The property is meant to be read-only in a way that you should be unable to replace the object with another one by a simple assignment. If you decide to return a mutable object from a property it's pretty similar to putting dictionary as an element of a tuple.
While I understand your point I'd probably go with something more explicit like a generator method returning 2-tuples containing param name and value. That'd leave using the values or creating a dictionary out of  them to the code using this value.

Having said that, I guess it's fine to leave it as is, although it makes object's behaviour a bit obscure.

Reply:
We should be adding docstrings at some point and enable pylint on this code.  I'd like to understand why you think there's a problem with returning a dictionary and why a generator with 2-tuples would be better, or any other kinds of concerns so that it can be clarified in the docstring or the implementation can be updated if it's causing some real issues.

I guess I'm seeing this from a C++ point of view and returning the object by value instead of by reference.  I really don't see how it makes a difference from a user point of view whether the attribute returns the reference of the dictionary (which is mutable) or a copy of it as it's exactly the same data structure.

Actually, I've just looked at this again and realised the arch params contain a dictionary for each architecture, so it should be a deep copy to achieve the intended purpose of keeping this read-only.